rcl_logging_interface/include/rcl_logging_interface/rcl_logging_interface.h

```diff
@@ -91,6 +95,59 @@ RCL_LOGGING_INTERFACE_PUBLIC
 RCUTILS_WARN_UNUSED
 rcl_logging_ret_t rcl_logging_external_set_logger_level(const char * name, int level);
 
+/// Get the logging directory.
+/**
+ * Uses various environment variables to construct a logging directory path.
+ *
+ * Use $ROS2_LOG_DIR if ROS2_LOG_DIR is set and not empty.
+ * Otherwise, use $ROS2_HOME/log, using ~/.ros for ROS2_HOME if not set or if empty.
+ *
+ * It also expands '~' to the current user's home directory.
+ *
+ * \param[inout] directory The C string pointer at which to write the directory path.
+ *   Only meaningful if the call is successful.
+ * \param[in] allocator The allocator to use for memory allocation.
+ * \return RCL_LOGGING_RET_OK if successful, or
+ * \return RCL_LOGGING_RET_ERROR if an unspecified error occurs.
+ */
+RCL_LOGGING_INTERFACE_PUBLIC
+inline
+rcl_logging_ret_t
+rcl_logging_get_logging_directory(const char ** directory, rcutils_allocator_t allocator)
+{
+  const char * log_dir_env;
+  const char * err = rcutils_get_env("ROS2_LOG_DIR", &log_dir_env);
+  if (!err && *log_dir_env != '\0') {
+    *directory = log_dir_env;
+  } else {
+    const char * ros2_home_dir_env;
+    err = rcutils_get_env("ROS2_HOME", &ros2_home_dir_env);
+    if (err || *ros2_home_dir_env == '\0') {
+      ros2_home_dir_env = rcutils_join_path("~", ".ros", allocator);
+      if (ros2_home_dir_env == NULL) {
+        return RCL_LOGGING_RET_ERROR;
+      }
+    }
+    *directory = rcutils_join_path(ros2_home_dir_env, "log", allocator);
+    if (*directory == NULL) {
+      return RCL_LOGGING_RET_ERROR;
+    }
+  }
+
+  // Expand home directory
+  if (SIZE_MAX != rcutils_find(*directory, '~')) {
+    const char * homedir = rcutils_get_home_dir();
+    if (homedir == NULL) {
+      return RCL_LOGGING_RET_ERROR;
+    }
+    *directory = rcutils_repl_str(*directory, "~", homedir, &allocator);
+    if (*directory == NULL) {
+      return RCL_LOGGING_RET_ERROR;
+    }
+  }
```

Comment:
(feel free to ignore)

I would rather have a function in `rcutils` for this ("expand path" or something like that).
It seems to be an useful tool on its own.

Reply:
Sure, I can do it and come back to modify this later on (since this PR already depends on a number of other PRs :sweat_smile:): https://github.com/ros2/rcutils/issues/293